File: .vscode/TicTacToeSubmission/TicTacToe.cpp
#include <iostream>
#include <cstring>

const int BOARD_SIZE = 3;

class TicTacToe {
public:
    TicTacToe() {
        InitializeBoard();
    }

    void PlayGame() {
        char currentPlayer = 'X';

        while (true) {
            DisplayBoard();
            std::cout << "Player " << currentPlayer << ", enter your move (e.g., A1): ";
            char input[3];
            std::cin >> input;

            if (strlen(input) != 2 || !isalpha(input[0]) || !isdigit(input[1])) {
                std::cout << "Invalid input. Please enter a valid move (e.g., A1)." << std::endl;
                continue;
            }

            int row = input[0] - 'A';
            int col = input[1] - '1';

            if (!IsLegalMove(row, col)) {
                std::cout << "Illegal move. Please choose an empty cell." << std::endl;
                continue;
            }

            board[row][col] = currentPlayer;

            if (CheckWin(currentPlayer)) {
                DisplayBoard();
                std::cout << "Player " << currentPlayer << " wins!" << std::endl;
                break;
            }

            bool isTie = true;
            for (int i = 0; i < BOARD_SIZE; i++) {
                for (int j = 0; j < BOARD_SIZE; j++) {
                    if (board[i][j] == ' ') {
                        isTie = false;
                        break;
                    }
                }
                if (!isTie) {
                    break;
                }
            }

            if (isTie) {
                DisplayBoard();
                std::cout << "It's a tie!" << std::endl;
                break;
            }

            currentPlayer = (currentPlayer == 'X') ? 'O' : 'X';
        }
    }

private:
    char board[BOARD_SIZE][BOARD_SIZE];

    void InitializeBoard() {
        for (int i = 0; i < BOARD_SIZE; i++) {
            for (int j = 0; j < BOARD_SIZE; j++) {
                board[i][j] = ' ';
            }
        }
    }

    void DisplayBoard() const {
        std::cout << "  1 2 3" << std::endl;
        for (int i = 0; i < BOARD_SIZE; i++) {
            std::cout << char('A' + i) << " ";
            for (int j = 0; j < BOARD_SIZE; j++) {
                std::cout << board[i][j];
                if (j < BOARD_SIZE - 1) {
                    std::cout << "|";
                }
            }
            std::cout << std::endl;
            if (i < BOARD_SIZE - 1) {
                std::cout << "  -+-+-" << std::endl;
            }
        }
    }

    bool IsLegalMove(int row, int col) const {
        return row >= 0 && row < BOARD_SIZE && col >= 0 && col < BOARD_SIZE && board[row][col] == ' ';
    }

    bool CheckWin(char player) const {
        for (int i = 0; i < BOARD_SIZE; i++) {
            if (board[i][0] == player && board[i][1] == player && board[i][2] == player) {
                return true; // Row win
            }
            if (board[0][i] == player && board[1][i] == player && board[2][i] == player) {
                return true; // Column win
            }
        }
        if (board[0][0] == player && board[1][1] == player && board[2][2] == player) {
            return true; // Diagonal win (top-left to bottom-right)
        }
        if (board[0][2] == player && board[1][1] == player && board[2][0] == player) {
            return true; // Diagonal win (top-right to bottom-left)
        }
        return false;
    }
};

int main() {
    while (true) {
        TicTacToe game;
        game.PlayGame();
        std::cout << "Play again? (y/n): ";
        char choice;
        std::cin >> choice;
        if (choice != 'y' && choice != 'Y') {
            break;
        }
    }

    return 0;
}

File: .vscode/TicTacToeSubmission/TictacToe.py
def print_board():
    for row in [board[i:i + 3] for i in range(0, 9, 3)]:
        print(" | ".join(row))
        print("-" * 9)

def check_win():
    win_conditions = [(0, 1, 2), (3, 4, 5), (6, 7, 8), (0, 3, 6), (1, 4, 7), (2, 5, 8), (0, 4, 8), (2, 4, 6)]
    for condition in win_conditions:
        if board[condition[0]] == board[condition[1]] == board[condition[2]] != " ":
            return True
    return False

def place_player(player, move):
    board[move] = player

def minimax(board, depth, is_maximizing):
    if check_win():
        if is_maximizing:
            return -1  # Opponent wins
        else:
            return 1  # Computer wins
    elif " " not in board:
        return 0  # It's a draw

    if is_maximizing:
        best = -10000
        for move in range(9):
            if board[move] == " ":
                place_player("O", move)  # Computer's move
                evaluation = minimax(board, depth - 1, False)
                board[move] = " "  # Undo the move
                best = max(best, evaluation)
        return best
    else:
        worst = 10000
        for move in range(9):
            if board[move] == " ":
                place_player("X", move)  # Player's move
                evaluation = minimax(board, depth - 1, True)
                board[move] = " "  # Undo the move
                worst = min(worst, evaluation)
        return worst

def find_best_move():
    best_move = -1
    best_eval = float('-inf')
    for move in range(9):
        if board[move] == " ":
            place_player("O", move)  # Computer's move
            eval = minimax(board, 0, False)
            board[move] = " "  # Undo the move
            if eval > best_eval:
                best_eval = eval
                best_move = move
    return best_move

board = [" " for _ in range(9)]
player = "X"
while True:
    print_board()
    if player == "X":
        try:
            move = int(input(f"Player {player}, enter your move (1-9): ")) - 1
            if board[move] == " ":
                place_player("X", move)
                if check_win():
                    print_board()
                    print(f"Player {player} wins!")
                    break
                elif " " not in board:
                    print_board()
                    print("It's a draw!")
                    break
                player = "O"
            else:
                print("Invalid move. Please try again.")
        except (ValueError, IndexError):
            print("Invalid input. Please enter a number between 1 and 9.")
    else:
        computer_move = find_best_move()
        place_player("O", computer_move)
        if check_win():
            print_board()
            print("Computer wins!")
            break
        elif " " not in board:
            print_board()
            print("It's a draw!")
            break
        player = "X"
